### DISCUSSION TITLE
Facet aggregation how to get all results

### DISCUSSION THREAD

**Discussion by RvZEU** [ravendb/ravendb#14349] [Q&A] [answered]
Hi,

i have created a Product model with basic information
`Product{
Name,
Brand,
Color
...
}`

i have succesfully created an index 

`from ProductsStores in docs.ProductsStores
select new {
    Name,
    Brand,
    Color,
    Sku,
    Ean
}`

and i can succesfully get all the facets from a query:
`from index 'facetedsearch' select facet(Brand), facet(Color)`

I would like to enable that the user can select multiple Brands when they have already selected a brand, when i do this:

`from index 'facetedsearch' where Brand='Nikon' select facet(Brand), facet(Color)`

the facet result of Brands will only be Nikon, is it possible to get all the results of Brands even though 'Nikon' is already selected, the Color facet should be based on the Brand selection.

**Comment by ayende** [comment, marked as answer]
The way to to do is to first issue the top level facets, and then remember them for the next queries as you narrow it down.

**Reply by RvZEU** [reply]
Thanx, this is indeed what i currently do.
